projects/packages/jetpack-mu-wpcom/src/features/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-launchpad.php

```diff
@@ -139,19 +139,23 @@ public function update_site_options( $request ) {
 		foreach ( $input as $key => $value ) {
 			switch ( $key ) {
 				case 'checklist_statuses':
-					$launchpad_checklist_tasks_statuses_option = (array) get_option( 'launchpad_checklist_tasks_statuses', array() );
-					$launchpad_checklist_tasks_statuses_option = array_merge( $launchpad_checklist_tasks_statuses_option, $value );
-
-					foreach ( $value as $task => $task_value ) {
-						if ( $task_value ) {
-							// If we're marking a task as complete, the value should be truthy, so fire the Tracks event.
-							wpcom_launchpad_track_completed_task( $task );
+					$complete_values   = array();
+					$incomplete_values = array();
+
+					foreach ( $value as $task_id => $task_status ) {
```

Comment:
Side note: should we verify that we have an array before we call the `foreach`?

Reply:
The `checklist_statuses` can only receive an array as value, it's guarded by the API definition: https://github.com/Automattic/jetpack/blob/1f0ccb72cdef7bacb3f0b112c249f5cf8b8ee4ea/projects/packages/jetpack-mu-wpcom/src/features/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-launchpad.php#L53

Reply:
That makes sense! Resolving.